nservicebus/messaging/routing.md

```diff
@@ -0,0 +1,165 @@
+
+```
+string proxy = "ProxyQueueName";
+busConfig.Routing().UnicastRoutingTable.AddStatic(typeof(OrderAccepted), new EndpointName("Sales"), proxy);
+```
+
+### Endpoint mapping
+
+The middle layer is responsible for mapping the endpoint name into the list of endpoint instance names. This mapping is done in a two-step process:
+ * Based on the endpoint name, find out what are all its instances
+ * Based on the message type and the list of instances, find out which instances should receive the message
+The default behavior of the second step is to select a single instance in a round-robin manner. A user can override it by registering a different `DistributionStrategy` (e.g. send to all instances if a cache synchronization protocol is required).
+
+#### Dynamic mapping
+
+To add a dynamic mapping rule you need to pass a function that takes an endpoint name and returns a collection of endpoint instance names. 
+
+```
+EndpointName sales = new EndpointName("Sales");
+busConfig.Routing().UnicastRoutingTable.AddDynamic(sales, e => new EndpointInstanceName[] {
+	new EndpointInstanceName(e, "1", "MachineA"),
+	new EndpointInstanceName(e, "2", "MachineB")
+});
+```
+
+In this example the rule returns two instances passing both user (`1`, `2`) and transport (`MachineA`, `MachineB`) discriminators.
+
+#### Static mapping
+
+When such a flexibility is not needed, the `KnownEndpoints` class provides a number of convenience methods that let you register a static mapping.
+
+```
+EndpointName sales = new EndpointName("Sales");
+busConfig.Routing().UnicastRoutingTable.AddStatic(sales, new EndpointInstanceName(sales, "1", null), new EndpointInstanceName(sales, "2", null));
+```
+
+You can also pass only the transport discriminators
+
+```
+EndpointName sales = new EndpointName("Sales");
+busConfig.Routing().UnicastRoutingTable.AddStaticUsingTransportDiscriminators(sales, "MachineA", "MachineB");
+```
+
+
+### Instance mapping
+
+The bottom layer is responsible for mapping between the instance name and a transport address. Usually it does not require intervention from the user as the selected transport automatically registers its translation rule. The only times where user is required to configure the instance mapping is when the default translation violates some naming rules implied by the transport infrastructure (e.g. the generated transport address is too long for a queue name in MSMQ). Different types of instance mapping have different importance.
+
+#### Exceptions
+
+User-provided exceptions have the highest importance, overriding the other sources. In order to register an exception you need to use following API:
```

Comment:
Reads funky :)
Is idea here to say that routes defined for faulted address translations have the highest importance?